diagnostics/TUTT/TUTT_calc_model.py

```diff
@@ -0,0 +1,648 @@
+#Jun 29 2021
+import matplotlib.pyplot as pyplot
+#from mpl_toolkits.basemap import Basemap
+#from mpl_toolkits.basemap import addcyclic
+import numpy as np
+from normalize import normalize
+# http://netcdf4-python.googlecode.com/svn/trunk/docs/netCDF4-module.html
+from netCDF4 import Dataset, num2date
+# http://scikit-image.org/
+from skimage import measure
+# conda install -c scitools shapely
+from shapely.geometry import Polygon, LineString
+from scipy.signal import argrelextrema
+from scipy.stats.stats import pearsonr
+from scipy.interpolate import interp2d
+from scipy.interpolate import RectBivariateSpline
+import cartopy.crs as ccrs
+import cartopy.mpl.ticker as cticker
+
+lat_coord = os.environ["lat_coord"]
+lon_coord = os.environ["lon_coord"]
+z200_var = os.environ["z200_var"]
+WK_DIR    = os.environ["WK_DIR"]
+CASENAME  = os.environ["CASENAME"]
+
+def add_cyclic_BL(dat,lon):
+  if dat.shape[2]!=lon.shape[0]:
+    print('Dimension error in the input data!')
+    return
+  dat1=np.zeros((dat.shape[0],dat.shape[1],dat.shape[2]+1))
+  lon1=np.zeros(lon.shape[0]+1)
+  dat1[:,:,:-1]=dat
+  dat1[:,:,-1]=dat[:,:,0]
+  lon1[:-1]=lon
+  lon1[-1]=lon[0]
+  return(dat1,lon1)
```

Comment:
Note for the framework crew - @wrongkindofdoctor and @tsjackson-noaa - we may want to standardize longitudinal operations such as this one inside the framework.